pkg/scheduler/objects/application.go

```diff
@@ -903,8 +901,21 @@ func (sa *Application) tryAllocate(headRoom *resources.Resource, nodeIterator fu
 		if sa.canReplace(request) {
 			continue
 		}
-		// resource must fit in headroom otherwise skip the request
+
+		// resource must fit in headroom otherwise skip the request (unless preemption could help)
 		if !headRoom.FitInMaxUndef(request.GetAllocatedResource()) {
+			// attempt preemption
+			if *preemptAttemptsRemaining > 0 {
+				*preemptAttemptsRemaining--
+				fullIterator := fullNodeIterator()
+				if fullIterator != nil {
```

Comment:
When is `fullIterator == nil`? Is this a legit case? I don't see coverage in the unit tests.

Reply:
Same as above.